Exception on /reset_password_request [GET]
Traceback (most recent call last):
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\gabri\Desktop\Microblog\app\routes.py", line 194, in reset_password_request
    send_password_reset_email(user)
  File "C:\Users\gabri\Desktop\Microblog\app\email.py", line 13, in send_password_reset_email
    token = user.get_reset_password_token()
AttributeError: 'function' object has no attribute 'get_reset_password_token'
Microblog startup
Exception on /reset_password_request [GET]
Traceback (most recent call last):
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\gabri\Desktop\Microblog\app\routes.py", line 194, in reset_password_request
    send_password_reset_email(user)
  File "C:\Users\gabri\Desktop\Microblog\app\email.py", line 13, in send_password_reset_email
    token = user.get_reset_password_token()
AttributeError: 'function' object has no attribute 'get_reset_password_token'
Microblog startup
Exception on /reset_password_request [GET]
Traceback (most recent call last):
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\gabri\Desktop\Microblog\app\routes.py", line 194, in reset_password_request
    send_password_reset_email(user)
  File "C:\Users\gabri\Desktop\Microblog\app\email.py", line 13, in send_password_reset_email
    token = user.get_reset_password_token()
AttributeError: 'function' object has no attribute 'get_reset_password_token'
Microblog startup
Exception on /reset_password_request [GET]
Traceback (most recent call last):
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\gabri\Desktop\Microblog\app\routes.py", line 194, in reset_password_request
    send_password_reset_email(user)
  File "C:\Users\gabri\Desktop\Microblog\app\email.py", line 13, in send_password_reset_email
    token = user.get_reset_password_token()
AttributeError: 'function' object has no attribute 'get_reset_password_token'
Microblog startup
Microblog startup
Exception on /reset_password_request [GET]
Traceback (most recent call last):
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\gabri\Desktop\Microblog\app\routes.py", line 197, in reset_password_request
    return render_template('reset_password_request.html', title='Reset Password', form=form)
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\jinja2\environment.py", line 1291, in render
    self.environment.handle_exception()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\jinja2\environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\gabri\Desktop\Microblog\app\templates\reset_password_request.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "C:\Users\gabri\Desktop\Microblog\app\templates\base.html", line 38, in top-level template code
    {% block content %} {% endblock %}
  File "C:\Users\gabri\Desktop\Microblog\app\templates\reset_password_request.html", line 9, in block 'content'
    {{ form.password.label }}<br>
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\jinja2\environment.py", line 474, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'app.forms.ResetPasswordRequestForm object' has no attribute 'password'
Microblog startup
Exception on /reset_password_request [POST]
Traceback (most recent call last):
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\gabri\Desktop\Microblog\app\routes.py", line 191, in reset_password_request
    user. User.query.filter_by(email=form.email.data).first()
AttributeError: 'function' object has no attribute 'User'
Microblog startup
Exception on /reset_password_request [POST]
Traceback (most recent call last):
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\gabri\Desktop\Microblog\venv\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\gabri\Desktop\Microblog\app\routes.py", line 194, in reset_password_request
    send_password_reset_email(user)
  File "C:\Users\gabri\Desktop\Microblog\app\email.py", line 13, in send_password_reset_email
    token = user.get_reset_password_token()
  File "C:\Users\gabri\Desktop\Microblog\app\models.py", line 64, in get_reset_password_token
    return jwt.encode(
AttributeError: module 'jwt' has no attribute 'encode'
Microblog startup
